server/migrations/1620672944865_update-triggers.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable camelcase */
+
+const { collections } = require('../db');
+
+exports.up = async (pgm) => {
+  await pgm.dropTrigger(collections.EMPLOYER_SITES, 'update_postal_code_s', { ifExists: true });
+  await pgm.createTrigger(collections.EMPLOYER_SITES, 'update_postal_code_s', {
+    when: 'BEFORE',
+    operation: ['INSERT', 'UPDATE OF body'],
+    language: 'plpgsql',
+    level: 'ROW',
+    replace: true,
+  },
+  `
+  DECLARE
+   loc record;
+   participant record;
+  BEGIN
+    IF NEW.body->>'postalCode' = OLD.body->>'postalCode' THEN
+      RETURN NEW;
+    END IF;
+
+    SELECT longitude::text,latitude::text INTO loc FROM geocodes WHERE REPLACE(postal_code, ' ', '')=REPLACE(NEW.body->>'postalCode', ' ', '');
+
+    IF NOT FOUND THEN
+      SELECT longitude::text,latitude::text INTO loc FROM geocodes WHERE SUBSTRING(REPLACE(postal_code, ' ', ''),1,5)=SUBSTRING(REPLACE(NEW.body->>'postalCode', ' ', ''),1,5);
+    END IF;
+
+    IF NOT FOUND THEN
+      IF NEW.body->>'location' IS NOT NULL THEN
+        NEW.body = NEW.body::jsonb - 'location';
+        DELETE FROM participants_distance where site_id = OLD.id;
+      END IF;
+      RETURN NEW;
+    END IF;
+
+    NEW.body = jsonb_set(NEW.body::jsonb, '{location}', ('{"type":"Point","coordinates":[' || loc.longitude || ',' || loc.latitude || ']}')::jsonb);
+    IF OLD IS NULL OR OLD.body->>'location' IS NULL THEN
```

Comment:
This is the case when a postal code is updated from being invalid to being valid.

We don't want to enter into the 'update' block because it relies on the records existing in the distance table ✔️